rpc/wrtc_call_queue_mongodb.go

```diff
@@ -571,18 +585,26 @@ func (queue *mongoDBWebRTCCallQueue) operatorLivenessLoop() {
 			})
 		}
 
-		if _, err := queue.operatorsColl.UpdateOne(queue.cancelCtx, bson.D{{webrtcOperatorIDField, queue.operatorID}}, bson.D{
+		result, err := queue.operatorsColl.UpdateOne(queue.cancelCtx, bson.D{{webrtcOperatorIDField, queue.operatorID}}, bson.D{
 			{
 				"$set",
 				bson.D{
 					{webrtcOperatorExpireAtField, time.Now().Add(operatorHeartbeatWindow)},
 					{webrtcOperatorHostsField, hostSizes},
 				},
 			},
-		}); err != nil {
+		})
+		if err != nil {
+			reason := "context_canceled"
 			if !errors.Is(err, context.Canceled) {
+				//nolint:goconst
+				reason = "other"
 				queue.logger.Errorw("failed to update operator document for self", "error", err)
 			}
+			operatorsCollUpdateFailures.Inc(queue.operatorID, reason)
+		} else if result.MatchedCount == 0 {
+			queue.logger.Errorw("no existing operator document found, could not update operator document")
+			operatorsCollUpdateFailures.Inc(queue.operatorID, "no_existing_operator_document")
```

Comment:
Yeah this would be interesting. No error log would be output, but the update might silently fail.